### PR TITLE
#BN-9 feat: Two factor authentication for vendors

### PR DESCRIPTION
**Pull request by EvelyneUmubyeyi**
# PR Description
In this PR, I created endpoints for sending an email to the user who logged in using email and password. The sent email contains TOTP (time-based one-time password) that lasts a few minutes. One endpoint sends the email, and the other endpoint is for verifying if the TOTP entered by user for two factor authentication is valid. This PR also includes tests and documentation for those endpoints.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] 2FA option is enabled on vendor login
- [x] Appropriate error messages are returned on unsuccessful attempts
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. Test the endpoints in [ swagger ] ( http://localhost:9090/api-docs/ )
4. run unit tests in terminal using ` npm run test `
# Screenshots (If appropriate/applicable)
image
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

**Review by uwdjany — DISMISSED** [review]

**Comment by Honore123** [comment]
Resolve conflicts!

File: src/database/models/orderItems.js
Comment (Honore123) [inline src/database/models/orderItems.js:1]: Use Prettier eslint extension in VSCode

**Review by uwdjany — DISMISSED** [review]
Well done!

File: src/middleware/verifyJWT.js
Comment (Honore123) [inline src/middleware/verifyJWT.js:1]: Translate response message in this file

File: src/utils/emailValidator.js
Comment (Honore123) [inline src/utils/emailValidator.js:1]: It will be better if you delete those comments

**Review by Honore123 — APPROVED** [review]

**Pull request merged**